packages/node/src/integrations/tracing/nest.ts

```diff
@@ -44,7 +54,140 @@ interface MinimalNestJsApp {
+    return moduleDef;
+  }
+
+  /**
+   * Wraps the @Injectable decorator.
+   */
+  private _getInjectableFileInstrumentation(versions: string[]): InstrumentationNodeModuleFile {
+    return new InstrumentationNodeModuleFile(
+      '@nestjs/common/decorators/core/injectable.decorator.js',
+      versions,
+      (moduleExports: { Injectable: InjectableTarget }) => {
+        if (isWrapped(moduleExports.Injectable)) {
+          this._unwrap(moduleExports, 'Injectable');
+        }
+        this._wrap(moduleExports, 'Injectable', this._createWrapInjectable());
+        return moduleExports;
+      },
+      (moduleExports: { Injectable: InjectableTarget }) => {
+        this._unwrap(moduleExports, 'Injectable');
+      },
+    );
+  }
+
+  /**
+   * Creates a wrapper function for the @Injectable decorator.
+   *
+   * Wraps the use method to instrument nest class middleware.
+   */
+  private _createWrapInjectable() {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    return function wrapInjectable(original: any) {
+      return function wrappedInjectable(options?: unknown) {
+        return function (target: InjectableTarget) {
+          // ensure class has not been patched before
+          if (target.sentryPatched) {
+            return original(options)(target);
+          } else {
+            Object.defineProperty(target, sentryPatched, { value: true });
+          }
+
+          // patch middleware
+          if (typeof target.prototype.use === 'function') {
+            const originalUse = target.prototype.use;
+
+            target.prototype.use = function (req: unknown, res: unknown, next: (error?: unknown) => void): void {
+              const prevSpan = getActiveSpan();
+
+              startSpanManual(
+                {
+                  name: target.name,
+                  attributes: {
+                    [SEMANTIC_ATTRIBUTE_SENTRY_OP]: 'middleware.nestjs',
+                    [SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN]: 'auto.middleware.nestjs',
+                  },
+                },
+                (span: Span) => {
+                  // patch  next to end span before next middleware is being called
+                  const wrappedNext = (error?: Error | unknown): void => {
+                    span.end();
+
+                    if (prevSpan) {
+                      withActiveSpan(prevSpan, () => {
+                        next(error);
+                      });
+                    } else {
+                      next(error);
+                    }
+                  };
+
+                  return originalUse.apply(this, [req, res, wrappedNext]);
+                },
+              );
+            };
+          }
+
+          return original(options)(target);
+        };
+      };
+    };
+  }
+}
+
+const instrumentNestCore = generateInstrumentOnce('Nest-Core', () => {
+  return new NestInstrumentation();
+});
+
+const instrumentMiddleware = generateInstrumentOnce('Nest-Middleware', () => {
```

Comment:
rename to be more generic because interceptors and guards will most likely also be covered by this implementation